### PR TITLE
fix build for different machine architecture

### PR DESCRIPTION
**Pull request by alipsidikp**
add a callback to generic next for arm64

### CODE REVIEW

File: x64tsc.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+//go:build amd64
```
Comment (kelindar) [inline x64tsc.go]: rename this to `next_amd64.go`

File: nox64tsc.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//go:build !amd64
+
+package xxrand
+
+const havex64tsc = false
+
+// https://stackoverflow.com/questions/27693145/rdtscp-versus-rdtsc-cpuid
+func x64tsc() uint64 {
```
Comment (kelindar) [inline nox64tsc.go]: This should contain the generic version 

```go
var epoch uint64

// Next returns the next epoch for the random, when called without an explicit
// epoch provided. This one simply reads a time stamp counter if available and
// uses it as the epoch.
func next() uint64 {
	return atomic.AddUint64(&epoch, 1)
}
```

File: x64tsc.go [hunk 3]
```diff
@@ -0,0 +1,8 @@
...
+
+package xxrand
+
+const havex64tsc = true
+
+// https://stackoverflow.com/questions/27693145/rdtscp-versus-rdtsc-cpuid
+func x64tsc() uint64
```
Comment (kelindar) [inline x64tsc.go]: I think we can rename this to `next() uint64`

File: nox64tsc.go [hunk 4]
```diff
@@ -0,0 +1,10 @@
+//go:build !amd64
```
Comment (kelindar) [inline nox64tsc.go]: rename this file to `next_generic.go`

File: x64tsc.go [hunk 5]
```diff
@@ -0,0 +1,8 @@
+//go:build amd64
+
+package xxrand
+
+const havex64tsc = true
```
Comment (kelindar) [inline x64tsc.go]: this flag is not needed I think

File: rand.go [hunk 6]
```diff
@@ -119,7 +115,7 @@ func genericNext() uint64 {
 
 func init() {
```
Comment (kelindar) [inline rand.go]: No need of this init() step

**Review by kelindar — CHANGES_REQUESTED** [review]

**Comment by alipsidikp** [comment]
@kelindar comments addressed

**Review by kelindar — APPROVED** [review]

**Pull request merged**